test/indexSpec.js

```diff
@@ -489,12 +489,106 @@ describe('PluginBells', function () {
       it('returns the plugin\'s prefix', function * () {
         yield assertResolve(this.plugin.getPrefix(), 'example.red.')
       })
+
+      it('fails without any prefix', function (done) {
+        const plugin = new PluginBells({
+          auth: {
+            // no prefix
+            account: 'http://red.example/accounts/mike',
+            password: 'mike'
+          }
+        })
+        plugin.getPrefix().catch((err) => {
+          assert.equal(err.message, 'Prefix has not been set')
+          done()
+        })
+      })
+
+      it('cannot connect without any prefix', function (done) {
+        const plugin = new PluginBells({
+          auth: {
+            account: 'http://blue.example/accounts/mike',
+            password: 'mike'
+          },
+          debugReplyNotifications: true,
+          debugAutofund: {
+            connector: 'http://mark.example',
+            admin: {username: 'adminuser', password: 'adminpass'}
+          }
+        })
+
+        const infoRedLedger = cloneDeep(require('./data/infoRedLedger.json'))
+        nock('http://blue.example')
+          .get('/')
+          .reply(200, infoRedLedger)
+
+        nock('http://blue.example')
+          .get('/accounts/mike')
+          .reply(200, {
+            ledger: 'http://blue.example',
+            name: 'mike'
+          })
+
+        plugin.connect().catch((err) => {
+          assert.equal(err.message, 'Unable to set prefix from ledger or from local config')
+          done()
+        })
+      })
+
+      it('gets the ledger\'s prefix when available', function * () {
+        const plugin = new PluginBells({
+          auth: {
+            prefix: 'example.foo.',
+            account: 'http://blue.example/accounts/mike',
+            password: 'mike'
+          },
+          debugReplyNotifications: true,
+          debugAutofund: {
+            connector: 'http://mark.example',
+            admin: {username: 'adminuser', password: 'adminpass'}
+          }
+        })
+
+        const infoRedLedger = Object.assign(
+          cloneDeep(require('./data/infoRedLedger.json')),
+          { ilp_prefix: 'example.blue.' }
+        )
+        nock('http://blue.example')
+          .get('/')
+          .reply(200, infoRedLedger)
+
+        nock('http://blue.example')
+          .get('/accounts/mike')
+          .reply(200, {
+            ledger: 'http://blue.example',
+            name: 'mike'
+          })
+
+        this.wsRedLedger = new wsHelper.Server('ws://blue.example/accounts/mike/transfers')
+
+        yield plugin.connect()
+        yield assertResolve(plugin.getPrefix(), 'example.blue.')
+        plugin.disconnect()
+      })
```

Comment:
When using `nock` it's good practice to call [`.done()`](https://github.com/node-nock/nock#expectations) at the end to make sure that it was actually called.